jsbits/isomorphic.js

```diff
@@ -1,20 +1,47 @@
-function copyDOMIntoVTree (vtree) {
-    walk (vtree, document.body.firstChild);
+function copyDOMIntoVTree(vtree) {
+  walk(vtree, document.body.firstChild);
 }
 
-function walk (vtree, node) {
-    var i = 0, vdomChild, domChild;
-    vtree.domRef = node;
-    while (i < vtree.children.length) {
-      vdomChild = vtree.children[i];
-      domChild = node.childNodes[i];
-      if (vdomChild.type === "vtext") {
-	  vdomChild.domRef = domChild;
-	  i++;
-	  continue;
+function walk(vtree, node) {
+  // This is slightly more complicated than one might expect since
+  // browsers will collapse consecutive text nodes into a single text node.
+  // There can thus be fewer DOM nodes than VDOM nodes.
+  var i = 0,
+    j = 0,
+    // This indicates if we are in a block of consecutive text nodes.
+    inTextBlock = false,
+    vdomChild,
+    domChild;
+  vtree.domRef = node;
+  for (i = 0; i < vtree.children.length; i++) {
+    vdomChild = vtree.children[i];
+    domChild = node.childNodes[j];
```

Comment:
I'm getting an error where `domChild` evaluates to `undefined` because the node has no children. Although I'm not sure my error is related to what's going on in this function (yet), I think we need to account for the possibility that `domChild` is undefined...

Reply:
That error message is what should be fixed by this PR so I don’t think we need to account for this.

Reply:
Hmm, so after a little more investigation the error I was seeing was due to the fact that my `View` started with a `body_` rather than with some internal element (like `div_` or whatever). Because of this, the `vtree` object in `copyDOMIntoVTree` was meant to represent the `body` tag instead of the first child of the `body` tag. So basically replacing `document.body.firstChild` with `document.body` fixed the issue. I'm not sure if that's in the scope of this PR, but I wanted to get it out there. Maybe we could do something like

```javascript
function copyDOMIntoVTree(vtree) {
  var elem = vtree.tag === "body" ? document.body : document.body.firstChild;
  walk(vtree, document.body); 
}
```

Unfortunately I don't have enough knowledge of exactly what the `walk` function is *supposed* to be doing, to know if that's the right move.

Reply:
It’s definitely out of scope for this PR but it’s also the wrong solution. You can only render inside of `body` with miso so this is a user error. We might consider outputting a warning in that case but trying to fix it is the wrong solution imho. Either way, let’s move discussions about things unrelated to this PR to separate issues.

Reply:
As of now we can render on any DOM element.